src/poetry/plugins/plugin_manager.py

```diff
@@ -80,4 +100,219 @@ def _load_plugin_entry_point(self, ep: metadata.EntryPoint) -> None:
+                missing_plugin_count -= 1
+                if missing_plugin_count == 0:
+                    break
+            else:
+                installed_packages.append(package)
+
+        if missing_plugin_count == 0 and not insufficient_plugins:
+            # all required plugins are installed and satisfy the requirements
+            self._write_config()
+            self._io.write_line(
+                "All required plugins have already been installed"
+                " in Poetry's environment."
+            )
+            self._io.write_line("")
+            return
+
+        if insufficient_plugins and self._io.is_debug():
+            plugins_str = "\n".join(
+                f"  - {req}\n    installed: {p}" for p, req in insufficient_plugins
+            )
+            self._io.write_line(
+                "The following Poetry plugins are required by the project"
+                f" but are not satisfied by the installed versions:\n{plugins_str}"
+            )
+
+        # install missing plugins
+        missing_plugins = [
+            plugin
+            for name, plugin in relevant_plugins.items()
+            if name not in satisfied_plugins
+        ]
+        plugins_str = "\n".join(f"  - {p}" for p in missing_plugins)
+        self._io.write_line(
+            "The following Poetry plugins are required by the project"
+            f" but are not installed in Poetry's environment:\n{plugins_str}\n"
+            f"Installing Poetry plugins only for the current project..."
+        )
+        self._install(missing_plugins, poetry_env, installed_packages)
+        self._io.write_line("")
+        self._write_config()
+
+    def _is_fresh(self) -> bool:
+        if not self._config_file.exists():
+            return False
+
+        with self._config_file.open("rb") as f:
+            stored_config = tomllib.load(f)
+
+        return stored_config == self._config
+
+    def _install(
+        self,
+        plugins: Sequence[Dependency],
+        poetry_env: Env,
+        locked_packages: Sequence[Package],
+    ) -> None:
+        project = ProjectPackage(name="poetry-project-instance", version="0")
+        project.python_versions = ".".join(str(v) for v in poetry_env.version_info[:3])
+        # consider all packages in Poetry's environment pinned
+        for package in locked_packages:
+            project.add_dependency(package.to_dependency())
+        # add missing plugin dependencies
+        for dependency in plugins:
+            project.add_dependency(dependency)
+
+        # force new package to be installed in the project cache instead of Poetry's env
+        poetry_env.paths["platlib"] = str(self._path)
+        poetry_env.paths["purelib"] = str(self._path)
+
+        # Build installed repository from locked packages so that plugins
+        # that may be overwritten are not included.
+        repo = Repository("poetry-repo")
+        for package in locked_packages:
+            repo.add_package(package)
```

Comment:
```suggestion
        repo = Repository("poetry-repo", locked_packages)
```
We can also eliminate that variable all together and put `Repository("poetry-repo", locked_packages)` directly in the `Installer` init in line 295.

Reply:
done